### ISSUE TITLE
All macos tests failing on master: Error installing bundler

### ISSUE THREAD

**Issue by jtattermusch** [grpc/grpc#17619]
```
Using /Users/kbuilder/.rvm/gems/ruby-2.5.0
ERROR:  Error installing bundler:
	The last version of bundler (>= 0) to support your Ruby & RubyGems was 1.17.3. Try installing it with `gem install bundler -v 1.17.3`
	bundler requires RubyGems version >= 3.0.0. The current RubyGems version is 2.7.8. Try 'gem update --system' to update RubyGems itself.
```

https://source.cloud.google.com/results/invocations/e06080fd-81c0-48db-87f8-023aa3ca00a8/log

Same for all other macos jobs as this problem is in the prepare script:
https://github.com/grpc/grpc/blob/8f5bda6de95f8948ab97084f2bbad6c6f5aec606/tools/internal_ci/helper_scripts/prepare_build_macos_rc#L45

**Comment by apolcyn** [comment]
The `bundler` ruby gem just released version https://rubygems.org/gems/bundler/versions/2.0.0 .... which drops support for ruby < 2.3 and rubygems < 3.0.0. I think these artifact builders need to be updated

**Comment by apolcyn** [comment]
Pinning bundler to 1.17.3 in these problematic places, in https://github.com/grpc/grpc/pull/17607

**Comment by apolcyn** [comment]
fixed by #17607
